src/ethereum_test_types/blob.py

```diff
@@ -0,0 +1,469 @@
+    FIELD_ELEMENTS_PER_BLOB,
+    KZG_ENDIANNESS,
+    # MAX_BLOB_GAS_PER_BLOCK,
+    # MAX_BLOBS_PER_BLOCK_ELECTRA,
+    # TARGET_BLOB_GAS_PER_BLOCK,
+    # TARGET_BLOBS_PER_BLOCK_ELECTRA,
+    Cancun,
+    Fork,
+    Osaka,
+    Prague,
+)
+
+TRUSTED_SETUP_FILE_NAME = "blob_trusted_setup.txt"
+TRUSTED_SETUP_PATH = Path(realpath(__file__)).parent / TRUSTED_SETUP_FILE_NAME
+TRUSTED_SETUP = ckzg.load_trusted_setup(str(TRUSTED_SETUP_PATH), 0)
+# print(f"{TRUSTED_SETUP_FILE_NAME} has been loaded")
+
+
+def get_eest_root_folder(marker_files=("pyproject.toml", ".git", "tests", "src")) -> Path:
+    """Search for a folder where all files/folders listed above exist (root of project)."""
+    current = Path(__file__).resolve()
+    for parent in current.parents:
+        if all((parent / marker).exists() for marker in marker_files):
+            return parent
+    raise RuntimeError("Project root folder of execution-spec-tests was not found")
+
+
+eest_root = get_eest_root_folder()
+
+
+class Blob(CamelModel):
+    """Class representing a full blob."""
+
+    data: Bytes
+    commitment: Bytes
+    proof: List[Bytes] | Bytes  # Bytes < Osaka, List[Bytes] >= Osaka
+    cells: List[Bytes] | None  # None (in json: null)  < Osaka, List[Bytes] >= Osaka
+
+    versioned_hash: Hash
+    name: str  # blob_<fork>_<seed>
+    fork: str
+    seed: int
+    timestamp: int
+
+    @staticmethod
+    def get_filepath(fork: str, seed: int, timestamp: int):
+        """Return the Path to the blob that would be created with these parameters."""
+        would_be_filename: str = "blob_" + fork + "_" + str(seed) + "_" + str(timestamp) + ".json"
+        assert fork in ["cancun", "prague", "osaka"], f"Fork {fork} not implemented yet"
+
+        if fork == "osaka":
+            would_be_static_blob_path: Path = (
+                eest_root
+                / "tests"
+                / "osaka"
+                / "eip7594_peerdas"
+                / "static_blobs"
+                / would_be_filename
+            )
+        elif fork == "prague":
+            would_be_static_blob_path = (
+                eest_root / "tests" / "prague" / "static_blobs" / would_be_filename
+            )
+        elif fork == "cancun":
+            would_be_static_blob_path = (
+                eest_root
+                / "tests"
+                / "cancun"
+                / "eip4844_blobs"
+                / "static_blobs"
+                / would_be_filename
+            )
+        return would_be_static_blob_path
+
+    @staticmethod
+    def NewBlob(fork: Fork, seed: int = 0, timestamp: int = 0) -> "Blob":  # noqa: N802
+        """Construct Blob instances. Fork-specific logic is encapsulated within nested functions."""  # noqa: E501
+        assert fork.engine_get_blobs_version() is not None, (
+            f"You provided fork {fork.name()} but it does not support blobs!"
+        )  # TODO: why does mypy say '"type[BaseFork]" has no attribute "engine_get_blobs_version"'?
```

Comment:
Can instead use `supports_blobs` here:
```suggestion
        assert fork.supports_blobs(), (
            f"Provided fork {fork.name()} but it does not support blobs!"
        )
```